src/main.cpp

```diff
@@ -3661,13 +3661,34 @@ bool CheckColdStakeFreeOutput(const CTransaction& tx, const int nHeight)
         if (lastOut.nValue == 3 * COIN)
             return true;
 
+        // This could be a budget block.
+        if (Params().IsRegTestNet())
+            return false;
+
+        // if mnsync is incomplete, we cannot verify if this is a budget block.
+        // so we check that the staker is not transferring value to the free output
+        if (!masternodeSync.IsSynced()) {
+            // First try finding the previous transaction in database
+            CTransaction txPrev; uint256 hashBlock;
+            if (!GetTransaction(tx.vin[0].prevout.hash, txPrev, hashBlock, true))
+                return error("%s : read txPrev failed: %s",  __func__, tx.vin[0].prevout.hash.GetHex());
+            CAmount amtIn = txPrev.vout[tx.vin[0].prevout.n].nValue + GetBlockValue(nHeight - 1);
```

Comment:
Why `nHeight - 1`?

Reply:
Because the block value is  usually get from `pindexPrev` (e.g. see `ConnectBlock` or `CBudgetManager::FillBlockPayee`, or `CMasternodePayments::FillBlockPayee`, or `CreateNewBlock` in miner). Also, check #967

Reply:
ok. Thanks for the clarification.